src/client/datascience/kernel-launcher/jupyterPaths.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import * as path from 'path';
+import { CancellationToken } from 'vscode';
+import { IPlatformService } from '../../common/platform/types';
+import { IPathUtils } from '../../common/types';
+import { IEnvironmentVariablesProvider } from '../../common/variables/types';
+import { tryGetRealPath } from '../common';
+
+const winJupyterPath = path.join('AppData', 'Roaming', 'jupyter', 'kernels');
+const linuxJupyterPath = path.join('.local', 'share', 'jupyter', 'kernels');
+const macJupyterPath = path.join('Library', 'Jupyter', 'kernels');
+export const baseKernelPath = path.join('share', 'jupyter', 'kernels');
+
+@injectable()
+export class JupyterPaths {
+    constructor(
+        @inject(IPlatformService) private platformService: IPlatformService,
+        @inject(IPathUtils) private readonly pathUtils: IPathUtils,
+        @inject(IEnvironmentVariablesProvider) private readonly envVarsProvider: IEnvironmentVariablesProvider
+    ) {}
+
+    /**
+     * This should return a WRITABLE place that jupyter will look for a kernel as documented
+     * here: https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs
+     */
+    public async getKernelSpecRootPath(): Promise<string | undefined> {
+        if (this.platformService.isWindows) {
+            return tryGetRealPath(path.join(this.pathUtils.home, winJupyterPath));
```

Comment:
I wish we knew exactly why we had to use `tryGetRealPythonpath`.
If we did, then everything in this class is a simple constant.

Reply:
On windows the path is not correct if you combine those variables. It won't point to a path that you can actually read from.

Reply:
Thanks, will add that comment